src/SilverpopConnector/SilverpopXmlConnector.php

```diff
@@ -706,6 +707,14 @@ public function updateRecipient($listId, $recipientId, $fields, $optParams=array
     foreach ($optParams as $key => $value) {
       $params .= "\t<{$key}>{$value}</{$key}>\n";
     }
+    if ($snoozeDate) {
+      $params.= '
+        <SNOOZE_SETTINGS>
+          <SNOOZED>true</SNOOZED>
+          <RESUME_SEND_DATE>' . $snoozeDate . '</RESUME_SEND_DATE>
+        </SNOOZE_SETTINGS>
+      ';
+    }
```

Comment:
Especially in methods that take an array of optional params, I find that for a case like this it's usually cleaner to detect a special case in the array rather than making it a new param.

```suggestion
    foreach ($optParams as $key => $value) {
      if ($key == 'snoozeDate') {
        $params .='<SNOOZE_SETTINGS><SNOOZED>true</SNOOZED><RESUME_SEND_DATE>' . $snoozeDate . '</RESUME_SEND_DATE></SNOOZE_SETTINGS>';
        continue;
      }
      $params .= "\t<{$key}>{$value}</{$key}>\n";
    }```

Reply:
Obviously if you decide to go this way you should clean up the docblock and the method signature to match...

Reply:
Hey thanks for the quick response - seems good to me!

Reply:
OK - I've pushed up the tidy up